tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/TraceAnnotations/TraceAnnotationsIntegration.cs

```diff
@@ -37,16 +37,28 @@ public class TraceAnnotationsIntegration
         internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, RuntimeMethodHandle methodHandle, RuntimeTypeHandle typeHandle)
         {
             var info = InstrumentedMethodCache.GetOrAdd(
-                    new RuntimeHandleTuple(methodHandle, typeHandle),
-                    key => TraceAnnotationInfoFactory.Create(MethodBase.GetMethodFromHandle(key.MethodHandle, key.TypeHandle)));
+                new RuntimeHandleTuple(methodHandle, typeHandle),
+                key => TraceAnnotationInfoFactory.Create(MethodBase.GetMethodFromHandle(key.MethodHandle, key.TypeHandle)));
 
+            return new CallTargetState(CreateSpan(info.ResourceName, info.OperationName, IntegrationId));
+        }
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <param name="resourceName">The resource name</param>
+        /// <param name="operationName">The operation name</param>
+        /// <param name="integrationId">The integration generating the span</param>
+        /// <returns>Calltarget state value</returns>
+        internal static Scope CreateSpan(string resourceName, string operationName, IntegrationId integrationId)
```

Comment:
```suggestion
        internal static Scope CreateScope(string resourceName, string operationName, IntegrationId integrationId)
```

Reply:
Done in 1bed3f68dfca764bd20a60fea21287b9274e0258.